### PR TITLE
Fix "Could not find method nexusStaging()"

### PR DESCRIPTION
**Pull request by PLNech**
This fixes a build error introduced by the recommended way to configure nexusStaging. 
See Codearte/gradle-nexus-staging-plugin#16 if you want more details.

### CODE REVIEW

File: algoliasearch/build.gradle [hunk 1]
```diff
@@ -21,7 +21,17 @@
  * THE SOFTWARE.
  */
 
+buildscript {
+    repositories {
+        jcenter()
```
Comment (szpak) [inline algoliasearch/build.gradle]: I would put here `mavenCentral()` - it is originally deployed to Maven Central and by the way mirrored to jcenter.

Comment (PLNech) [inline algoliasearch/build.gradle]: Good point, thanks!

**Comment by clement-leprovost-algolia** [comment]
@PLNech So if I get it right, you introduced the Nexus plugin to automate the publication to Maven Central, but this introduces a dependency that is not available in some contexts, making the build fail? And this PR fixes it?

File: algoliasearch/build.gradle [hunk 2]
```diff
@@ -33,10 +33,11 @@ ext {
     - Log-in on sonatype.org, go to staging profile, select com.algolia, id = last part of URL */
 }
 
-
-nexusStaging {
-    packageGroup = PUBLISH_GROUP_ID
-    stagingProfileId = PUBLISH_STAGING_ID
+if (project.hasProperty("nexus")) {
```
Comment (szpak) [inline algoliasearch/build.gradle]: Btw, have it failed when it was put in the root `build.gradle` (I have not experience with Android apps)?

**Comment by PLNech** [comment]
@clement-leprovost-algolia: Exactly, the default setup required any project to add it in its root `build.gradle` to compile. 
With this fix we only apply the plugin conditionally with the *nexus* property, as in [here](https://github.com/algolia/algoliasearch-client-android/pull/59/files#diff-e55bb7573f1110c8b2a6922fe8cccf48L39), when releasing.

File: algoliasearch/build.gradle (hunk 2 above)
Comment (PLNech) [inline algoliasearch/build.gradle]: @szpak: Oh good point, moving the `nexusStaging` closure to the root `build.gradle` works and is simpler!

Comment (szpak) [inline algoliasearch/build.gradle]: Great. This plugin does not do much when just applying and not executed. I don't know if there is much gain when surrounded with an `if` block (and there is no warning as it is applied on the root project).

Comment (PLNech) [inline algoliasearch/build.gradle]: Right. Anyway, I moved the configuration to the root `build.gradle` and everything's fine :)

**Pull request merged**